packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -89,23 +95,34 @@ export default function __experimentalUseColors(
 	const createComponent = useMemo(
 		() =>
 			memoize(
-				( property, color, colorValue, customColor ) => ( { children } ) =>
+				( name, property, className, color, colorValue, customColor ) => ( {
+					children,
+					className: componentClass = '',
+					style: componentStyles = {},
+				} ) =>
 					// Clone children, setting the style property from the color configuration,
 					// if not already set explicitly through props.
 					Children.map( children, ( child ) => {
-						let className = child.props.className;
-						let style = child.props.style;
+						const elementClassName = classnames(
+							componentClass,
+							child.props.className,
+							{
+								[ `has-${ kebabCase( color ) }-${ kebabCase( property ) }` ]: color,
```

Comment:
```suggestion
								[ `has-${ kebabCase( color ) }-${ kebabCase( name ) }` ]: color,
```

Shouldn't we also use the `name` here?

Reply:
No, here we use kebabCase( property ), e.g: we use has-secondary-color not has-secondary-text-color.

Reply:
But, we use `has-text-color` in the other one? That's confusing, is there a reason it was done this way?